ui/app/components/PolicyComponent/PolicyComponent.js

```diff
@@ -228,61 +80,37 @@ export default class PolicyComponent extends React.Component {
     } else {
       w = {
         wType,
-        [wKey]: {...work[0][wKey], [attribute]: value}
+        [wKey]: {...worksWithWType[0][wKey], [attribute]: value}
       }
     }
-    if (Object.values(w[wKey]).every((el) => isNaN(el))) { //Remove work type element from work array if all of the attributes are empty
-      this.setState((prevState) => ({
-        json: {
-          version: prevState.json.version,
-          policyDetails: {
-            ...prevState.json.policyDetails, policy: {
-              ...prevState.json.policyDetails.policy, work: [...prevWorks.filter((w) => {
-                return w.wType !== wType
-              })]
-            }
-          },
-        }
-      }));
-    } else {
-      this.setState((prevState) => ({
-          json: {
-            version: prevState.json.version,
-            policyDetails: {
-              ...prevState.json.policyDetails, policy: {
-                ...prevState.json.policyDetails.policy, work: [...prevWorks.filter((w) => {
-                  return w.wType !== wType
-                }), w]
-              }
-            },
-          }
-        })
-      );
+    const works = [...prevWorks.filter((w) => {
+      return w.wType !== wType
+    })];
+    if (!Object.values(w[wKey]).every((el) => isNaN(el))) { //Add work type element from work array if any of the attributes are non empty
+      works.push(w);
     }
+    this.props.policyWorkChangeAction(works);
   }
 
-  handleSubmitClick(e) {
-    if ((this.state.query.type === 'POST' || this.state.query.type === 'PUT') && this.state.query.state === 'PENDING') {
-      this.setState({
-        msg: "Please wait, your previous policy change is still pending"
-      });
+  handleSubmitClick() {
+    if ((this.props.versionedPolicyDetails.reqType === 'POST' || this.props.versionedPolicyDetails.reqType === 'PUT') && this.props.versionedPolicyDetails.asyncStatus === 'PENDING') {
+      document.querySelector('#policy-submit').MaterialSnackbar.showSnackbar({message: 'Please wait, your previous policy change is still pending'});
```

Comment:
can you replace these with a helper method? too  hard on eyes

Reply:
Replaced with the helper method.